src/helpers/when.js

```diff
@@ -0,0 +1,6 @@
+'use strict'
+
+const isFunction = (input) => typeof input === 'function';
+
+export default (predicate) => (elemOrThunk) =>
```

Comment:
im not sure about this 'when' function, why to use it? how can improve our performance/code using it?

Reply:
Just to keep the logic of the component above the return and the view inside. No performance improvment